### PR TITLE
exporting group parameters from Revit

### PR DESCRIPTION
**Pull request by samberger**
## Description & motivation

Currently the parameters of Revit's model groups aren't exported to Speckle at all. Similar to other built elements, the added code aims to add the parameters property object to the exported speckle element.

[as dicussed in the Speckle Community Forum](https://speckle.community/t/revits-built-in-parameter-all-model-type-name-saves-null/8256/12?u=samberger)

## Changes:

- added function `AddGroupParamsToDict()` to `ConversionUtils.cs`
- added parameters export to `GroupToSpeckle()` in `ConvertGroup.cs`

## To-do before merge:

- Unfortunately, I wasn't able to test the code myself, but most of it was copied from other converters and adapted accordingly for groups. So maybe do an extra double check. Sorry.

## Screenshots:

The exported data for `groups` before:

grafik

There should be an additional `parameters{}` property after.

## Checklist:

- [ ] My pull request follows the guidelines in the [Contributing guide](https://github.com/specklesystems/speckle-server/blob/main/CONTRIBUTING.md)?
- [x] My pull request does not duplicate any other open [Pull Requests](../../pulls) for the same update/change?
- [x] My commits are related to the pull request and do not amend unrelated code or documentation.
- [x] My code follows a similar style to existing code.
- [ ] I have added appropriate tests.
- [ ] I have updated or added relevant documentation.

### CODE REVIEW

**Comment by bimgeek** [comment]
Hey @samberger ,

Thanks for the PR. I have scheduled a ticket to get this in as part of our next sprint (starting Wednesday).

**Comment by connorivy** [comment]
Hey @samberger,

This functionality makes a lot of sense so it would be good to include it. I have a couple questions though. The "AddGroupParamsToDict" seems (at a glance) to be identitcal to our existing "AddElementParamsToDict" method. What is the reason that this method is needed? And similarly, the code that you added to the group conversion method seems similar to existing code as well.

Typically, we would just add the line "GetAllRevitParamsAndIds()" to the conversion. Does that not work for groups?

**Comment by samberger** [comment]
Hi @connorivy,

Yes, most is copy/pasted to a separate function in order to avoid unintended effects on how it works for other elements, since I'm not really into the rest of the code and couldn't get a feeling for that.

But you are completely right! I had another look, and I just noticed that `Group` actually extends `Element`, so it should work the same way... Makes sense of course, but I guess I just haven't seen it.

That's why I thought I would need a separate function and also copied anything else, which seem to apply for groups, from `GetAllRevitParamsAndIds()` directly into `GroupToSpeckle()` 

Long story short, the PR was more thought of as a proposal (starting to get into the connectors' code), and you're right.. so do it the way you proposed and don't merge. :) I'll already be happy about being able to export the group parameters.

Thanks! :)

**Comment by connorivy** [comment]
@samberger 

Perfect. Thanks for the push on this. I have made a beautiful one-line PR https://github.com/specklesystems/speckle-sharp/pull/3293 that will get included in 2.19
